diff --git a/gio/gunixvolume.c b/gio/gunixvolume.c
index 981f703..48ea9fd 100644
--- a/gio/gunixvolume.c
+++ b/gio/gunixvolume.c
@@ -110,7 +110,8 @@ _g_unix_volume_new (GVolumeMonitor  *volume_monitor,
   GUnixVolume *volume;
   
   if (!(g_unix_mount_point_is_user_mountable (mountpoint) ||
-	g_str_has_prefix (g_unix_mount_point_get_device_path (mountpoint), "/vol/")) ||
+	g_str_has_prefix (g_unix_mount_point_get_device_path (mountpoint), "/vol/") ||
+	g_str_has_prefix (g_unix_mount_point_get_device_path (mountpoint), "/dev/mmcblk1")) ||
       g_unix_mount_point_is_loopback (mountpoint))
     return NULL;
   
