diff --git a/glib/gmain.c b/glib/gmain.c
index 227168a0f..850638735 100644
--- a/glib/gmain.c
+++ b/glib/gmain.c
@@ -2349,7 +2349,7 @@ g_source_remove (guint tag)
   if (source)
     g_source_destroy (source);
   else
-    g_critical ("Source ID %u was not found when attempting to remove it", tag);
+    g_warning ("Source ID %u was not found when attempting to remove it", tag);
 
   return source != NULL;
 }
