diff --git a/glib/gmain.c b/glib/gmain.c
index 29ccee0..79e4fa6 100644
--- a/glib/gmain.c
+++ b/glib/gmain.c
@@ -2281,7 +2281,7 @@ g_source_remove (guint tag)
   if (source)
     g_source_destroy (source);
   else
-    g_critical ("Source ID %u was not found when attempting to remove it", tag);
+    g_warning ("Source ID %u was not found when attempting to remove it", tag);
 
   return source != NULL;
 }
